.github/workflows/macos-generation.yml

```diff
@@ -0,0 +1,212 @@
+name: macOS image generation
+on:
+  workflow_call:
+    inputs:
+      image_label:
+        type: string
+        description: macOS codename
+        required: true
+      base_image_name:
+        type: string
+        description: Base clean image
+        required: true
+      template_path:
+        type: string
+        description: Packer template path
+        required: true
+      target_datastore:
+        type: string
+        description: Image datastore
+        required: true
+      custom_repo:
+        type: string
+        description: Custom repo to checkout
+        required: false
+      custom_repo_commit_hash:
+        type: string
+        description: Custom repo commit hash
+        required: false
+
+env:
+  KEYVAULT: imagegeneration
+  ESXI_CLUSTER: mcv2-build-unstable
+  VCENTER_DATACENTER: imagegen
+  OUTPUT_FOLDER: mms-output
+  BUILD_DATASTORE: ds-image
+
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  build:
+    runs-on: maccloudv2
+    timeout-minutes: 1200
```

Comment:
we should change the agent tags, lets pick `azure-ubuntu` for ubuntu builds `azure-windows` for windows ones and `macos-vmware` for current macs.

@Alexey-Ayupov @maxim-lobanov ideas?

Reply:
I believe the tag should represent the environment where it locates. I suggest `agent-vmware` and `agent-azure` or something similar.